packages/gallery/src/lib/dom.js

```diff
@@ -0,0 +1,102 @@
+import { sanitise, updateCSS } from './utils';
+import { ATTRIBUTE, EVENTS, KEYCODES } from './constants';
+
+export const init = store => () => {
+    const state = store.getState();
+    const { settings, dom, list } = state;
+
+    if (!dom.liveRegion) console.warn(`A live region announcing current and total items is recommended for screen readers.`);
+    else writeLiveRegion(state);
+    /* istanbul ignore next */
+    if (dom.fullscreen) {
+        if (document.fullscreenEnabled || document.webkitFullscreenEnabled) {
+            dom.fullscreen.addEventListener('click', toggleFullScreen.bind(null, store));
+            document.addEventListener('fullscreenchange', e => {
+                if (document.fullscreenElement) document.documentElement.classList.add(settings.className.fullscreen);
+                else document.documentElement.classList.remove(settings.className.fullscreen);
+            });
+        } else dom.fullscreen.parentNode.removeChild(dom.fullscreen);
+    }
+    if (dom.previous) dom.previous.addEventListener('click', previous.bind(null, store));
+    if (dom.next) dom.next.addEventListener('click', next.bind(null, store));
+    if (dom.triggers.length) {
+        dom.triggers.forEach(trigger => trigger.addEventListener('click', () => {
+            goTo(store)(+trigger.getAttribute(ATTRIBUTE.NAVIGATE));
+        }));
+    }
+
+    /* istanbul ignore next */
+    list.addEventListener('keydown', e => {
+        switch (e.keyCode) {
+        case KEYCODES.LEFT:
+            e.preventDefault();
+            previous(store);
+            break;
+        case KEYCODES.RIGHT:
+            e.preventDefault();
+            next(store);
+            break;
+        // case KEYCODES.TAB:
+        //     next(store);
+        //     break;
+        }
+    });
+
+    updateCSS(state);
+    broadcast(store, EVENTS.INITIALISED)(state);
+};
+
+const writeLiveRegion = ({ activeIndex, items, settings, dom }) => dom.liveRegion.innerHTML = sanitise(settings.announcement(activeIndex + 1, items.length));
+
+/* istanbul ignore next */
+export const toggleFullScreen = store => {
+    const { node } = store.getState();
+    if (!document.fullscreenElement) {
+        if (node.requestFullscreen) node.requestFullscreen();
+        else if (node.webkitRequestFullscreen) node.webkitRequestFullscreen();
+    } else {
+        if (document.exitFullscreen) document.exitFullscreen();
+        else if (document.webkitExitFullscreen) document.webkitExitFullscreen();
+    }
+};
+
+export const change = (store, next, options = { fromListener: false, fromScroll: false }) => {
+    const { list, items, settings } = store.getState();
+
+    store.update({ ...store.getState(), activeIndex: next }, [
+        () => {
+            if (!options.fromScroll) list.scrollLeft = (next * items[next].clientWidth);
+        },
+        writeLiveRegion,

```

Comment:
Have we maybe lost the line that dealt with focus here?  https://github.com/stormid/components/blob/3cb615ed7c6a0965d9ad4bb7bb130185df74aea0/packages/gallery/src/lib/dom.js#L88 

I noticed when trying the buttons in NVDA, because focus stays on the next/prev buttons in this version, we're only getting '2 of 4' read out, rather than the details of the new slide we got in the previous one.